src/Cardano/Wallet/Api/Types.hs

```diff
@@ -67,13 +67,13 @@ import qualified Data.Aeson.Types as Aeson
 -------------------------------------------------------------------------------}
 
 data Wallet = Wallet
-    { _id :: !WalletId
+    { _id :: !(ApiT WalletId)
```

Comment:
is there a reason we are wrapping all in `ApiT` that I am not aware off?

Or we explicitly want to keel all json instances defined in this module?

aha - we probably want to enforce the same `genericToJSON` and `genericParseJSON` rules for the whole api, so we are going to wrap most/all things with `ApiT`?

Reply:
-> Or we explicitly want to keel all json instances defined in this module?

You said it :) 
A nice explanation from @jonathanknowles here: https://github.com/input-output-hk/cardano-wallet/pull/107#discussion_r268021219